### PR TITLE
avoid infinite loop with invalid objects

### PR DESCRIPTION
**Pull request by Ekleog**
Actually check for NONSENSICAL_SIZE when in DriverMode::Forced, and introduce a regression test.

### CODE REVIEW

**Comment by camshaft** [comment]
Good catch! Thanks!

File: lib/bolero-generator/src/arbitrary.rs [hunk 1]
```diff
@@ -80,4 +80,22 @@ mod tests {
...
+            if v >= 1024 {
+                return Err(arbitrary::Error::IncorrectFormat);
+            }
+            Ok(UnlikelyToBeValid(v))
+        }
+    }
+
+    #[test]
+    fn unlikely_to_be_valid() {
+        let _ = generator_test!(gen_arbitrary::<UnlikelyToBeValid>());
```
Comment (camshaft) [inline lib/bolero-generator/src/arbitrary.rs:99]: @Ekleog I saw a failure on this test: https://github.com/camshaft/bolero/actions/runs/8257285194/job/22587562228?pr=208#step:9:522

We'll need to figure out how to make it less flaky.

Comment (Ekleog) [reply lib/bolero-generator/src/arbitrary.rs:99]: Hmm is the random number generation not uniform? 1024 / 2**128 is definitely not something that should in any way be flaky :sweat_smile:

Comment (camshaft) [reply lib/bolero-generator/src/arbitrary.rs:99]: yeah it should be uniform... i haven't had a chance to dig into why it failed

**Pull request merged**